src/index.ts

```diff
@@ -0,0 +1,212 @@
+async function execCommands(commands: string[]) {
+  if (!commands.length) {
+    return;
+  }
+
+  startGroup("🚀 Pre/Post Commands Group");
+
+  const arrPromises = commands.map(async (command) => {
+    const npxCMD = command.startsWith("wrangler") ? `npx ${command}` : command;
+
+    info(`🚀 Executing command: ${npxCMD}`);
+
+    await exec(npxCMD, {
+      cwd: config["workingDirectory"],
+      env: process.env,
+    });
+  });
+
+  await Promise.all(arrPromises);
+  endGroup();
+}
+
+async function uploadSecrets() {
+  const secrets = config["secrets"];
+  if (!secrets.length) {
+    return;
+  }
+
+  startGroup("🔑 Uploading Secrets");
+  const npxCMD = process.env.RUNNER_OS === "Windows" ? "npx.cmd exec" : "npx";
+  const environment = config["ENVIRONMENT"];
+
+  const getSecret = (secret: string) =>
+    process.env[secret] ?? setFailed("🚨 Secret not found in env");
+
+  const environmentSuffix = !environment.length ? "" : ` --env ${environment}`;
+
+  await Promise.all(
+    secrets.map(async (secret) =>
+      await exec(
+        `${npxCMD} wrangler secret put ${getSecret(
+          secret
+        )}${environmentSuffix}`,
+        {
+          cwd: config["workingDirectory"],
+          env: process.env,
+          stdio: "ignore",
+        }
+      )
+    )
+  );
+
+  info(`✅ Uploaded secrets`);
+  endGroup();
+}
+
+function defaultCommandBehavior() {
+  const environment = config["ENVIRONMENT"];
+  const wranglerVersion = config["WRANGLER_VERSION"];
+  const vars = config["VARS"];
+  const workingDirectory = config["workingDirectory"];
+
+  const deployCommand =
+    wranglerVersion === "latest" || wranglerVersion.startsWith("3")
+      ? "deploy"
+      : "publish";
+
+  info(`📌 No Wrangler commands were provided, executing default deployment. '${deployCommand}'`);
+
+  const envVarArray = vars.map((envVar: string) => {
+    if (process.env[envVar] && process.env[envVar]?.length !== 0) {
+      return `${envVar}:${process.env[envVar]!}`;
+    } else {
+      setFailed(`🚨 ${envVar} not found in variables.`);
+    }
+  });
+
+  const envVarArg: string =
+    envVarArray.length > 0 ? `--var ${envVarArray.join(" ").trim()}` : "";
+
+  if (environment.length === 0) {
+    spawnSync(`npx wrangler ${deployCommand} ${envVarArg}`.trim(), {
+      cwd: workingDirectory,
+      env: process.env,
+    });
+  } else {
+    spawnSync(
+      `npx wrangler ${deployCommand} --env ${environment} ${envVarArg}`.trim(),
+      { cwd: workingDirectory, env: process.env }
+    );
+  }
+}
+
+async function wranglerCommands() {
+  startGroup("🚀 Executing Wrangler Commands");
+  const commands = config["COMMANDS"];
+  if (!commands.length) {
+    defaultCommandBehavior();
+    return;
+  }
+  const environment = config["ENVIRONMENT"];
+
+  if (environment.length > 0 && commands.every((command) => command.includes("--env"))) {
+    setFailed(
+      `🚨 An environment as been specified adding '--env ${environment}' is required in the command.`
+    );
+  }
```

Comment:
```suggestion
  if (environment.length > 0 && !commands.every((command) => command.includes(`--env ${environment}`))) {
    setFailed(
      `🚨 Since an environment has been specified, each command must include '--env ${environment}'.`
    );
  }
```

Reply:
oops my logic was backward on that .every lol